src/core/interpreter.cc

```diff
@@ -256,11 +256,13 @@ void SetGlobalArrayInternal(lua_State* lua, const char* name, Interpreter::Slice
 /* In case the error set into the Lua stack by PushError() was generated
  * by the non-error-trapping version of redis.pcall(), which is redis.call(),
  * this function will raise the Lua error so that the execution of the
- * script will be halted. */
-int RaiseError(lua_State* lua) {
+ * script will be halted.
+ * This function never returns, it unwinds the Lua call stack until an error handler is found or the
+ * script exits */
+void RaiseError(lua_State* lua) {
```

Comment:
Consider renaming this function to something that SHOUTS that it does not return. Maybe let it look like a macro with capital letters.
`RAISE_ERROR_AND_RETURN()` maybe?